File: src/main/java/com/algorithms/codechef/roadmap/CheckPalindrome.java
package com.algorithms.codechef.roadmap;

import java.util.Scanner;

public class CheckPalindrome {

    class Main {
        // Function to check if the string can be a palindrome after deleting at most
        // one character
        static boolean validPalindrome(String s) {
            // Write your code here
            int r = s.length() - 1;
            int l = 0;
            boolean life = true;

            while (l < r) {
                if (s.charAt(l) != s.charAt(r) && life) {
                    if (s.charAt(l + 1) == s.charAt(r)) {
                        life = false;
                        l++;
                    } else {
                        life = false;
                        r--;
                    }
                } else if (s.charAt(l) != s.charAt(r)) {
                    return false;
                } else {
                    r--;
                    l++;
                }
            }
            return true;
        }

        public static void main(String[] args) {
            Scanner scanner = new Scanner(System.in);
            int n = scanner.nextInt();
            String s = scanner.next();
            System.out.println(validPalindrome(s) ? "true" : "false");
        }
    }
}

File: src/main/cpp/codechef/recursion/check_palindrome.cpp
#include <iostream>
using namespace std;

string checkPalindrome(string s, int n) {
    //Write your code here
    if (n <= 1) return "Yes";
    if (s.at(n-1) != s.at(s.size()-n)) return "No";
    return checkPalindrome(s, n-1);
}

int main() {
    string s;
    cin >> s;
    cout << checkPalindrome(s, s.size()) << endl;
    return 0;
}
